sopel/run_script.py

```diff
@@ -148,19 +149,22 @@ def build_parser():
 
 
 def check_not_root():
+
     """Check if root is running the bot.
 
     It raises a ``RuntimeError`` if the user has root privileges on Linux or
     if it is the ``Administrator`` account on Windows.
     """
-    try:
+    if platform.system() in ["Linux", "Darwin"]:
```

Comment:
You could use a temporary variable in order to reuse its value in exception (in the last `else` section):

```
name = platform.system()
if name in [...]:
    # doing fine
else:
    raise RuntimeError('Error: Could not detect Operating Sytem Type, or it is unsupported: %s' % name)
```

Reply:
got this one fixed

Reply:
re: https://github.com/sopel-irc/sopel/pull/1477#discussion_r255975860, you should be able to  add "Solaris" to this list